oqamaint/openqabot.py

```diff
@@ -0,0 +1,588 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+
+from collections import namedtuple
+from datetime import date
+import md5
+from pprint import pformat
+import re
+
+
+import requests
+import osc.core
+
+import ReviewBot
+from osclib.comments import CommentAPI
+
+from suse import SUSEUpdate
+
+try:
+    from xml.etree import cElementTree as ET
+except ImportError:
+    from xml.etree import ElementTree as ET
+
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+
+QA_UNKNOWN = 0
+QA_INPROGRESS = 1
+QA_FAILED = 2
+QA_PASSED = 3
+
+Package = namedtuple('Package', ('name', 'version', 'release'))
+pkgname_re = re.compile(r'(?P<name>.+)-(?P<version>[^-]+)-(?P<release>[^-]+)\.(?P<arch>[^.]+)\.rpm')
+comment_marker_re = re.compile(
+    r'<!-- openqa state=(?P<state>done|seen)(?: result=(?P<result>accepted|declined|none))?(?: revision=(?P<revision>\d+))? -->')
+
+
+class OpenQABot(ReviewBot.ReviewBot):
+
+    """ check ABI of library packages
+    """
+
+    def __init__(self, target_repo, project_settings, api_map, *args, **kwargs):
+
+        self.tgt_repo = target_repo
+        self.project_settings = project_settings
+        self.api_map = api_map
+        super(OpenQABot, self).__init__(*args, **kwargs)
+
+        self.force = False
+        self.openqa = None
+        self.commentapi = CommentAPI(self.apiurl)
+        self.update_test_builds = {}
+        self.pending_target_repos = set()
+        self.openqa_jobs = {}
+
+    def gather_test_builds(self):
+        for prj, u in self.tgt_repo[self.openqa.baseurl].items():
+            buildnr = 0
+            cjob = 0
+            for j in self.jobs_for_target(u):
+                # avoid going backwards in job ID
+                if cjob > int(j['id']):
+                    continue
+                buildnr = j['settings']['BUILD']
+                cjob = int(j['id'])
+            self.update_test_builds[prj] = buildnr
+            jobs = self.jobs_for_target(u, build=buildnr)
+            self.openqa_jobs[prj] = jobs
+            if self.calculate_qa_status(jobs) == QA_INPROGRESS:
+                self.pending_target_repos.add(prj)
+
+    # reimplemention from baseclass
+    def check_requests(self):
+
+        if self.apiurl.endswith('.suse.de'):
```

Comment:
There already is:

https://github.com/openSUSE/osc-plugin-factory/blob/4011159df44ce05b3c74a53e29ffe007d1b27a26/ReviewBot.py#L72

Reply:
nice , thx